container/kvm/initialisation.go

```diff
@@ -29,7 +40,24 @@ func NewContainerInitialiser() container.Initialiser {
 
 // Initialise is specified on the container.Initialiser interface.
 func (ci *containerInitialiser) Initialise() error {
-	return ensureDependencies()
+	if err := ensureDependencies(); err != nil {
+		return errors.Trace(err)
+	}
+
+	// Check if we've done this already.
+	poolInfo, err := poolInfo(run)
```

Comment:
I don't think so. The worker/provisioner.ContainerSetup.runInitialiser method calls Initialise() in a mutex. This allows it to be called more than once (idempotently) if there is an error after createPool.  Let me know if you read otherwise.